test/test_gpu2.cpp

```diff
@@ -7296,6 +7296,44 @@ TEST_F(NVFuserTest, FusionPredicateElimination8_CUDA) {
       "T6 should not be predicated");
 }
 
+TEST_F(NVFuserTest, FusionPredicateElimination9_CUDA) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const int M = 1024, split = 512;
+
+  // Algorithm
+  auto tv0 = makeContigTensor(1, DataType::Float);
+  auto tv1 = set(tv0);
+
+  fusion->addInput(tv0);
+  fusion->addOutput(tv1);
+
+  // Schedule
+  auto tv0c = tv0->cacheAfter();
+  tv0c->setMemoryType(MemoryType::Shared);
+  tv0c->axis(-1)->parallelize(ParallelType::TIDx);
+
+  tv1->split(-1, split);
+  tv1->axis(-1)->parallelize(ParallelType::TIDx);
+
+  // Validation
+  at::manual_seed(0);
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto t0 = at::randn({M}, options);
+
+  {
+    GpuLower gpulw(fusion.get());
+    TORCH_CHECK(PredicatedChecker::isPredicated(tv0c, gpulw));
```

Comment:
Why is `tv0c` is predicated? I am expecting that it is not because the predicate should be a trivial `threadIdx.x < T0.size[0]`.

Reply:
Could you search our codebase and find all `isExact` usage with `lower_utils::isExtentEqualParallelTypeExact`?

Reply:
BTW, running the test with `NVFUSER_DUMP="lower_verbose"` will provide helpful debugging information on which pass are these predicates generated.

Reply:
Thanks for early feedback!

I created this PR to have a clean foundation. When I created the test I noticed that it passes when all tv are expected to be predicated, which is unexpected - I plan to debug this further.